jsonrpc/src/common/messageReader.ts

```diff
@@ -215,15 +215,20 @@ export class ReadableStreamMessageReader extends AbstractMessageReader {
 				if (!headers) {
 					return;
 				}
-				const contentLength = headers.get('Content-Length');
-				if (!contentLength) {
-					throw new Error('Header must provide a Content-Length property.');
-				}
-				const length = parseInt(contentLength);
-				if (isNaN(length)) {
-					throw new Error('Content-Length value must be a number.');
+				try {
+					const contentLength = headers.get('Content-Length');
+					if (!contentLength) {
+						throw new Error('Header must provide a Content-Length property.');
```

Comment:
Instead of throwing here and then catching again we can directly call `this.fireError(...)`. 

We should also but the reader into a state where it doesn't attempt to read any messages anymore since it will not be able to advance in this situation.